extension/src/ALObject/ALParser.ts

```diff
@@ -109,52 +109,50 @@ function parseProcedureDeclaration(
     const attributes: string[] = [];
     let lineNo = procedureLineNo - 1;
     let loop = true;
-    do {
-      const line = alCodeLines[lineNo].code;
-      const attributeMatch = line.match(attributePattern);
-      if (attributeMatch) {
-        if (attributeMatch.groups?.attribute) {
-          attributes.push(attributeMatch[0].trim());
+    if (lineNo > 0) {
+      do {
+        const line = alCodeLines[lineNo];
+        const attributeMatch = line.code.match(attributePattern);
+        if (attributeMatch) {
+          if (attributeMatch.groups?.attribute) {
+            attributes.push(attributeMatch[0].trim());
+          }
+        } else if (!line.isInsignificant()) {
+          loop = false;
         }
-      } else {
-        const ignoreRegex = new RegExp(ignoreCodeLinePattern, "im");
-        // console.log(ignoreCodeLinePattern); // Comment out
-        const ignoreMatch = line.match(ignoreRegex);
-        if (!ignoreMatch) {
+        lineNo--;
+        if (lineNo < 0) {
           loop = false;
         }
```

Comment:
if you decide to keep the current loop structure.
```suggestion
        loop = lineNo < 0;
```

Reply:
Nope, `loop` is set in another condition as well